debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java

```diff
@@ -0,0 +1,134 @@
+            + "  NLS_TIMESTAMP_TZ_FORMAT = 'YYYY-MM-DD HH24:MI:SS.FF TZH:TZM'"
+            + "  NLS_NUMERIC_CHARACTERS = '.,'";
+
+    /**
+     * This returns statement to build log miner view for online redo log files
+     * @param startScn mine from
+     * @param endScn mine till
+     * @param strategy Log Mining strategy
+     * @return statement todo: handle corruption. STATUS (Double) — value of 0 indicates it is executable
+     */
+    static String getStartLogMinerStatement(Long startScn, Long endScn, OracleConnectorConfig.LogMiningStrategy strategy, boolean isContinuousMining) {
+        String miningStrategy;
+        if (strategy.equals(OracleConnectorConfig.LogMiningStrategy.CATALOG_IN_REDO)) {
+            miningStrategy = "DBMS_LOGMNR.DICT_FROM_REDO_LOGS + DBMS_LOGMNR.DDL_DICT_TRACKING ";
+        }
+        else {
+            miningStrategy = "DBMS_LOGMNR.DICT_FROM_ONLINE_CATALOG ";
+        }
+        if (isContinuousMining) {
+            miningStrategy += " + DBMS_LOGMNR.CONTINUOUS_MINE ";
+        }
+        return "BEGIN sys.dbms_logmnr.start_logmnr(" +
+                "startScn => '" + startScn + "', " +
+                "endScn => '" + endScn + "', " +
+                "OPTIONS => " + miningStrategy +
+                " + DBMS_LOGMNR.NO_ROWID_IN_STMT);" +
+                "END;";
+    }
+
+    /**
+     * This is the query from the log miner view to get changes. Columns of the view we using are:
+     * NOTE. Currently we do not capture changes from other schemas
+     * SCN - The SCN at which a change was made
+     * COMMIT_SCN - The SCN at which a change was committed
+     * USERNAME - Name of the user who executed the transaction
+     * SQL_REDO Reconstructed SQL statement that is equivalent to the original SQL statement that made the change
+     * OPERATION_CODE - Number of the operation code.
+     * TABLE_NAME - Name of the modified table
+     * TIMESTAMP - Timestamp when the database change was made
+     *
+     * @param schemaName user name
+     * @param logMinerUser log mining session user name
+     * @param schema schema
+     * @return the query
+     */
+    static String queryLogMinerContents(String schemaName, String logMinerUser, OracleDatabaseSchema schema) {
+        List<String> whiteListTableNames = schema.tableIds().stream().map(TableId::table).collect(Collectors.toList());
+
+        String sorting = "ORDER BY SCN";
+        return "SELECT SCN, SQL_REDO, OPERATION_CODE, TIMESTAMP, XID, CSF, TABLE_NAME, SEG_OWNER, OPERATION, USERNAME " +
+                " FROM " + LOGMNR_CONTENTS_VIEW + " WHERE  OPERATION_CODE in (1,2,3,5) " + // 5 - DDL
+                " AND SEG_OWNER = '" + schemaName.toUpperCase() + "' " +
+                buildTableInPredicate(whiteListTableNames) +
+                " AND SCN >= ? AND SCN < ? " +
+                " OR (OPERATION_CODE IN (5,7,34,36) AND USERNAME NOT IN ('SYS','SYSTEM','" + logMinerUser.toUpperCase() + "'))" + sorting; // todo username = schemaName?
```

Comment:
34 as well? MISSING_SCN. Interesting.

Reply:
@AndreyIg do you have any input on this?

Reply:
By doing this I was intended to address a few non-standard situations. There a re more on todos

Reply:
Needs tracking, @Naros?

Reply:
Logged as [DBZ-2530](https;//issues.redhat.com/browse/DBZ-2530).